system/ad_service_state_monitor/src/ad_service_state_monitor_node/ad_service_state_monitor_node.cpp

```diff
@@ -164,17 +171,24 @@ bool AutowareStateMonitorNode::onShutdownService(
   const std::shared_ptr<std_srvs::srv::Trigger::Response> response)
 {
   (void)request_header;
-  state_input_.is_finalizing = true;
+
+  {
+    std::lock_guard<std::mutex> lock(lock_state_input_);
+    state_input_.is_finalizing = true;
+  }
 
   const auto t_start = this->get_clock()->now();
   constexpr double timeout = 3.0;
   while (rclcpp::ok()) {
     // rclcpp::spin_some(this->get_node_base_interface());
 
-    if (state_machine_->getCurrentState() == AutowareState::Finalizing) {
-      response->success = true;
-      response->message = "Shutdown Autoware.";
-      return true;
+    {
+      std::lock_guard<std::mutex> lock(lock_state_machine_);
```

Comment:
@ytakano 
Could you change to use lock and  unlock correctly like[ behavior path planner](https://github.com/autowarefoundation/autoware.universe/blob/b23f4188b26f21ba1f4443be1d1e6819a1747f69/planning/behavior_path_planner/src/behavior_path_planner_node.cpp#L514) inorder not to change previous code much with fix PR?

Reply:
I think `std::lock_guard` is safer than lock/unlock because `std::lock_guard` automatically unlock a mutex.
I recommend to use `std::lock_guard` instead of lock/unlock in the behavior path planner.

Reply:
I think @taikitanaka3  says the lock is only needed under if condition and should be released before following line `response->success = true`.

Reply:
However, in this case, it returns immediately, so I think it's OK.

Reply:
yes but I want to limit the effective range of lock guard as small as possible

Reply:
Thank you!
I wii fix it.